sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -2515,10 +2515,12 @@ export interface SearchSuggester {
 }
 
 // @public
-export type SelectFields<T extends object> = T extends Array<infer Elem> ? Elem extends object ? SelectFields<Elem> : never : {
-    [Key in keyof T]: Key extends string ? NonNullable<T[Key]> extends object ? NonNullable<T[Key]> extends ExcludedODataTypes ? Key : SelectFields<NonNullable<T[Key]>> extends infer NextPaths ? NextPaths extends string ? // Union this key with all the next paths separated with '/'
-    Key | `${Key}/${NextPaths}` : Key : never : Key : never;
-}[keyof T & string] & string;
+export type SelectArray<Fields = never> = [string] extends [Fields] ? Fields[] | Readonly<Fields[]> : (<T>() => T extends Fields ? true : false) extends <T>() => T extends never ? true : false ? string[] | Readonly<string[]> : Fields[] | Readonly<Fields[]>;
```

Comment:
`Fields[] | Readonly<Fields[]>` sticks out a little bit. Could you help me understand why this is necessary? Will only `readonly Fields[]` work?

Reply:
I believe this is a holdover from when `SearchPick` wasn't homomorphic. It looks like all of the (hopefully sufficiently thorough) regression testing holds up fine when I change all of these `T[] | Readonly<T[]>` signatures to `readonly T[]`, so I'm glad to have that small readability win. Thanks for catching it!